packages/wallet/src/redux/protocols/transaction-submission/reducer.ts

```diff
@@ -1,30 +1,34 @@
 import {
   TransactionAction,
-  TRANSACTION_SENT_TO_METAMASK,
+  TRANSACTION_SENT,
   TRANSACTION_SUBMISSION_FAILED,
   TRANSACTION_SUBMITTED,
   TRANSACTION_CONFIRMED,
-  RETRY_TRANSACTION,
+  TRANSACTION_FAILED,
+  TRANSACTION_RETRY_APPROVED,
+  TRANSACTION_RETRY_DENIED,
 } from './actions';
 import {
   TransactionSubmissionState as TSState,
   waitForSubmission,
   approveRetry,
   waitForConfirmation,
   success,
-  start,
-  START,
+  waitForSend,
+  WAIT_FOR_SEND,
   WAIT_FOR_SUBMISSION,
   WAIT_FOR_CONFIRMATION,
   APPROVE_RETRY,
+  failure,
 } from './states';
-import { TransactionRequest } from 'ethers/providers';
 import { unreachable } from '../../../utils/reducer-utils';
 import { SharedData } from '..';
+import { TransactionRequest } from 'ethers/providers';
+import { queueTransaction } from '../../state';
 
 type Storage = SharedData;
 
-interface ReturnVal {
+export interface ReturnVal {
```

Comment:
I'm not a huge fan of exporting something called`ReturnVal`. It seems a bit generic.